app/Repositories/Cts/StudentRepository.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Repositories\Cts;
+
+use App\Models\Cts\PositionValidation;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Support\Collection;
+
+class StudentRepository
+{
+    public function getStudentsWithin(int $rtsId): Collection
+    {
+        $positionValidations = PositionValidation::with(['member', 'position'])
+        ->whereHas('position', function (Builder $query) use ($rtsId) {
+            $query->where('rts_id', '=', $rtsId);
+        })->where('status', '=', 1)->get();
+
+        $students = collect();
+
+        foreach ($positionValidations as $positions) {
+            $students->push($positions->member);
+        }
+
+        return $this->format($students->unique());
+    }
+
+    private function format(Collection $data)
```

Comment:
Seems similar across repositories - possibly worth convert to helper function or a base repository type class? Or even a trait if you're feeling sacrilegious on this 'holy' day

Reply:
A helper wouldn't really "help" as the format is likely to be different. A base repository class could work but I can see scenarios in which we don't really need to format the return type... Not sure. Maybe. Don't think it is hurting as is?